libs/libc/pthread/pthread_mutexattr_setprotocol.c

```diff
@@ -54,21 +54,24 @@ int pthread_mutexattr_setprotocol(FAR pthread_mutexattr_t *attr,
   linfo("attr=0x%p protocol=%d\n", attr, protocol);
   DEBUGASSERT(attr != NULL);
 
-#ifdef CONFIG_PRIORITY_INHERITANCE
   if (protocol >= PTHREAD_PRIO_NONE && protocol <= PTHREAD_PRIO_PROTECT)
```

Comment:
·
can simplify to:
···
switch (protocol)
{
case PTHREAD_PRIO_NONE:
#ifdef CONFIG_PRIORITY_INHERITANCE
case PTHREAD_PRIO_INHERIT:
#endif
    attr->proto = protocol;
    return OK;

case PTHREAD_PRIO_PROTECT:
   return ENOTSUP;

default:
  return EINVAL;
}
```
}

Reply:
This will not return `ENOTSUP` in case if priority inheritance is disabled

Reply:
also `attr->proto` is not accessible if `CONFIG_PRIORITY_INHERITANCE=n`

Reply:
how about:
```
switch (protocol)
  {
    case PTHREAD_PRIO_NONE:
#ifdef CONFIG_PRIORITY_INHERITANCE
    case PTHREAD_PRIO_INHERIT:
      attr->proto = protocol;
#endif
      return OK;

    case PTHREAD_PRIO_PROTECT:
      return ENOTSUP;

    default:
      return EINVAL;
  }
```

Reply:
I think maybe only next:
```
  switch (protocol)
    {
      case PTHREAD_PRIO_NONE:
#ifdef CONFIG_PRIORITY_INHERITANCE
      case PTHREAD_PRIO_INHERIT:
        attr->proto = protocol;
        return OK;
#else
        return OK;

      case PTHREAD_PRIO_INHERIT:
#endif

      case PTHREAD_PRIO_PROTECT:
        return ENOTSUP;

      default:
        return EINVAL;
    }
```
But IMO this is ugly.

Reply:
this one?
```
if (protocol < PTHREAD_PRIO_NONE || protocol > PTHREAD_PRIO_PROTECT)
  {
     return EINVAL;
  }

switch (protocol)
  {
    case PTHREAD_PRIO_NONE:
#ifdef CONFIG_PRIORITY_INHERITANCE
    case PTHREAD_PRIO_INHERIT:
      attr->proto = protocol;
#endif
      return OK;

    default:
      return ENOTSUP;
  }
```

Reply:
Yes. This one should work fine.

Reply:
@xiaoxiang781216 anyway I do not see any benefit in such rework I will integrate a proposed change

Reply:
Done